bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,663 @@
+    private static final Pattern DATE_REGEX = Pattern
+            .compile("[A-Za-z]{3}\\s[A-Za-z]{3}\\s\\d\\d\\s\\d\\d\\d\\d\\s\\d\\d:\\d\\d:\\d\\d\\sGMT[-+]\\d\\d\\d\\d");
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("EEE MMM dd yyyy HH:mm:ss 'GMT'Z");
+    private static final Pattern DECIMAL_REGEX = Pattern.compile("-?\\d+\\.\\d+");
+    private static final String ASSET_FILE_PREFIX = "/remoteassets/remote_asset";
+    private static final Set<String> PROTECTED_PROPERTIES = new HashSet<>(Arrays.asList(
+            JcrConstants.JCR_CREATED, JcrConstants.JCR_CREATED_BY, JcrConstants.JCR_VERSIONHISTORY, JcrConstants.JCR_BASEVERSION,
+            JcrConstants.JCR_ISCHECKEDOUT, JcrConstants.JCR_UUID, JcrConstants.JCR_PREDECESSORS
+    ));
+    private static final Set<String> PROTECTED_NODES = new HashSet<>(Arrays.asList(
+            DamConstants.THUMBNAIL_NODE, AccessControlConstants.REP_POLICY
+    ));
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    @Reference
+    private RemoteAssetsConfig remoteAssetsConfig;
+
+    @Reference
+    private DynamicClassLoaderManager dynamicClassLoaderManager;
+
+    private ResourceResolver resourceResolver;
+    private Session session;
+    private ValueFactory valueFactory;
+    private int saveRefreshCount = 0;
+
+    /**
+     * Method to run on activation.
+     * @throws RepositoryException exception
+     */
+    @Activate
+    protected void activate() throws RepositoryException {
+        this.resourceResolver = RemoteAssets.logIn(this.resourceResolverFactory);
+        this.session = this.resourceResolver.adaptTo(Session.class);
+        if (StringUtils.isNotBlank(remoteAssetsConfig.getEventUserData())) {
+            this.session.getWorkspace().getObservationManager().setUserData(this.remoteAssetsConfig.getEventUserData());
+        }
+        this.valueFactory = this.session.getValueFactory();
+    }
+
+    /**
+     * Method to run on deactivation.
+     */
+    @Deactivate
+    protected void deactivate() {
+        if (this.session != null) {
+            try {
+                this.session.logout();
+            } catch (Exception e) {
+                LOG.warn("Failed session.logout()", e);
+            }
+        }
+
+        if (this.resourceResolver != null) {
+            try {
+                this.resourceResolver.close();
+            } catch (Exception e) {
+                LOG.warn("Failed resourceResolver.close()", e);
+            }
+        }
+    }
+
+    /**
+     * @see RemoteAssetsNodeSync#syncAssetNodes()
+     */
+    @Override
+    public void syncAssetNodes() {
+        List<String> syncPaths = new ArrayList<>();
+        syncPaths.addAll(this.remoteAssetsConfig.getTagSyncPaths());
+        syncPaths.addAll(this.remoteAssetsConfig.getDamSyncPaths());
+
+        try {
+            for (String syncPath : syncPaths) {
```

Comment:
So this is what creates the "remove asset nodes"... is this ONLY getting assets that directly reside in the syncPath folder? or can this feature descend into the folder sub-tree?

Reply:
I assume you mean "remote" asset nodes.  Yes, this is what syncs the nodes in.  Code is recursive, thus traversing the entire node tree downward one folder at a time.